src/main/scala/uk/gov/nationalarchives/tdr/api/auth/AuthorisationTags.scala

```diff
@@ -114,3 +114,16 @@ object ValidateUserOwnsFiles extends AuthorisationTag {
       })
   }
 }
+
+object ValidateHasAntiVirusMetadataAccess extends SyncAuthorisationTag {
+  override def validateSync(ctx: Context[ConsignmentApiContext, _]): BeforeFieldResult[ConsignmentApiContext, Unit] = {
+    val token = ctx.ctx.accessToken
+    val antivirusAccess = token.backendChecksRoles.contains("anti_virus")
+
+    if (antivirusAccess) {
+      continue
+    } else {
+      throw AuthorisationException("No access to anti-virus metadata")
```

Comment:
Just a thought: is there any useful info in the token, e.g. a user ID or something, that we could include in the logs to work out what client was trying to access the metadata? It would probably have to be something that's in a regular user token and a service client token.

Reply:
Added the user id for the Keycloak generated service account user to the message. This should allow tracing back to the client etc.